From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Nicolas Iooss <nicolas.iooss_linux@m4x.org>
Date: Sat, 14 Apr 2018 15:53:18 +0200
Subject: [PATCH] {For LLVMLinux} Make asm goto support optional

clang does not support asm goto yet.

https://bugs.llvm.org/show_bug.cgi?id=9295

This partially reverts commit d0266046ad54 ("x86: Remove
FAST_FEATURE_TESTS")
---
 arch/x86/Makefile                 | 2 +-
 arch/x86/include/asm/cpufeature.h | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/arch/x86/Makefile b/arch/x86/Makefile
index 39e823b1d749..a6becdf5dcca 100644
--- a/arch/x86/Makefile
+++ b/arch/x86/Makefile
@@ -181,7 +181,7 @@ ifdef CONFIG_FUNCTION_GRAPH_TRACER
 endif
 
 ifndef CC_HAVE_ASM_GOTO
-  $(error Compiler lacks asm-goto support.)
+  $(warning Compiler lacks asm-goto support.)
 endif
 
 #
diff --git a/arch/x86/include/asm/cpufeature.h b/arch/x86/include/asm/cpufeature.h
index aced6c9290d6..93860bf99b7c 100644
--- a/arch/x86/include/asm/cpufeature.h
+++ b/arch/x86/include/asm/cpufeature.h
@@ -147,7 +147,7 @@ extern void clear_cpu_cap(struct cpuinfo_x86 *c, unsigned int bit);
  * headers, but clang does not support ASM GOTO and fails the build.
  */
 #ifndef __BPF_TRACING__
-#warning "Compiler lacks ASM_GOTO support. Add -D __BPF_TRACING__ to your compiler arguments"
+/* #warning "Compiler lacks ASM_GOTO support. Add -D __BPF_TRACING__ to your compiler arguments" */
 #endif
 
 #define static_cpu_has(bit)            boot_cpu_has(bit)
-- 
